File: src/main/java/reai/timetracker/entity/Employee.java
package reai.timetracker.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "employees")
public class Employee {
    @Id
    private Long id; // Same as ReAI employee ID

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String email;

    private String department;

    @Column(name = "tenant_id")
    private Long tenantId;

    public Employee() {}

    public Employee(Long id, String name, String email) {
        this.id = id;
        this.name = name;
        this.email = email;
    }

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getDepartment() { return department; }
    public void setDepartment(String department) { this.department = department; }

    public Long getTenantId() { return tenantId; }
    public void setTenantId(Long tenantId) { this.tenantId = tenantId; }

    public String getDisplayName() {
        return name + " (" + email + ")";
    }
}

File: src/main/kotlin/reai/timetracker/entity/Employee.kt
package reai.timetracker.entity

import jakarta.persistence.*
import java.io.Serializable

@Entity
@Table(name = "employees")
data class Employee(


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    var id: Long = -1,

    @Column(nullable = false)
    var name: String = "",

    @Column(nullable = false, unique = true)
    var email: String = "",

) : Serializable {
    val displayName: String
        get() = "$name ($email)"
}

data class EmployeesDto(
    val id: Long?,
    val name: String?,
)
